src/location.rs

```diff
@@ -106,11 +106,11 @@ impl SrcLocation {
     }
 
     /// Returns this location as `DstLocation`
-    pub fn to_dst(&self) -> DstLocation {
+    pub fn to_dst(self) -> DstLocation {
```

Comment:
Should this consume self?

Reply:
Yea I think it can, we use them in `sn_routing` and `sn_node`, to avoid the creation of new objects. Otherwise, we need to change the name of the message mentioning it returns a new DstLocation.